patches/roslyn-analyzers/0001-WIP-for-source-build.patch

```diff
@@ -0,0 +1,1087 @@
+     -->
+-    <PackageId>*$(MSBuildProjectFullPath)*</PackageId>
++    <PackageId>*$(MSBuildProjectFullPath.Replace('/', '\\'))*</PackageId>
+   </PropertyGroup>
+   
+   <ItemGroup>
+@@ -28,4 +28,4 @@
+     <ProjectReference Include="..\..\src\Microsoft.CodeAnalysis.Analyzers\CSharp\Microsoft.CodeAnalysis.CSharp.Analyzers.csproj" />
+     <ProjectReference Include="..\..\src\Microsoft.CodeAnalysis.Analyzers\VisualBasic\Microsoft.CodeAnalysis.VisualBasic.Analyzers.vbproj" />
+   </ItemGroup>
+-</Project>
+\ No newline at end of file
++</Project>
+diff --git a/src/Microsoft.CodeAnalysis.AnalyzerUtilities/Microsoft.CodeAnalysis.AnalyzerUtilities.csproj b/src/Microsoft.CodeAnalysis.AnalyzerUtilities/Microsoft.CodeAnalysis.AnalyzerUtilities.csproj
+index 5bc31228b..721c84823 100644
+--- a/src/Microsoft.CodeAnalysis.AnalyzerUtilities/Microsoft.CodeAnalysis.AnalyzerUtilities.csproj
++++ b/src/Microsoft.CodeAnalysis.AnalyzerUtilities/Microsoft.CodeAnalysis.AnalyzerUtilities.csproj
+@@ -6,7 +6,7 @@
+       PackageId is used by Restore. If we set it to Microsoft.CodeAnalysis.AnalyzerUtilities,
+       Restore would conclude that there is a cyclic dependency between us and the Microsoft.CodeAnalysis.AnalyzerUtilities package.
+     -->
+-    <PackageId>*$(MSBuildProjectFullPath)*</PackageId>
++    <PackageId>*$(MSBuildProjectFullPath.Replace('/', '\\'))*</PackageId>
+     <ExcludeInternalFlowAnalyses>true</ExcludeInternalFlowAnalyses>
+     <ExcludeCodeMetricsUtilities>true</ExcludeCodeMetricsUtilities>
+     <ReleaseTrackingOptOut>true</ReleaseTrackingOptOut>
+diff --git a/src/Microsoft.CodeAnalysis.Analyzers/Core/Microsoft.CodeAnalysis.Analyzers.csproj b/src/Microsoft.CodeAnalysis.Analyzers/Core/Microsoft.CodeAnalysis.Analyzers.csproj
+index eab29554e..a2d208329 100644
+--- a/src/Microsoft.CodeAnalysis.Analyzers/Core/Microsoft.CodeAnalysis.Analyzers.csproj
++++ b/src/Microsoft.CodeAnalysis.Analyzers/Core/Microsoft.CodeAnalysis.Analyzers.csproj
+@@ -6,7 +6,7 @@
+       PackageId is used by Restore. If we set it to Microsoft.CodeAnalysis.Analyzers 
+       Restore would conclude that there is a cyclic dependency between Microsoft.CodeAnalysis and Microsoft.CodeAnalysis.Analyzers.
+     -->
+-    <PackageId>*$(MSBuildProjectFullPath)*</PackageId>
++    <PackageId>*$(MSBuildProjectFullPath.Replace('/', '\\'))*</PackageId>
+   </PropertyGroup>
+   <ItemGroup>
+     <InternalsVisibleTo Include="Microsoft.CodeAnalysis.CSharp.Analyzers" />
+@@ -18,4 +18,4 @@
+   </ItemGroup>
+   <Import Project="..\..\Utilities\Compiler\Analyzer.Utilities.projitems" Label="Shared" />
+   <Import Project="..\..\Utilities\Workspaces\Workspaces.Utilities.projitems" Label="Shared" />
+-</Project>
+\ No newline at end of file
++</Project>
+diff --git a/src/Microsoft.CodeAnalysis.Analyzers/Setup/AnalyzerReleases.Shipped.md b/src/Microsoft.CodeAnalysis.Analyzers/Setup/AnalyzerReleases.Shipped.md
+new file mode 100644
```

Comment:
A standalone build of `roslyn-analyzers` was failing for me with messages like:

> /home/omajid/devel/dotnet/roslyn-analyzers/src/Directory.Build.targets(45,5): error : Create a new empty file named 'AnalyzerReleases.Shipped.md' in the project directory [/home/omajid/devel/dotnet/roslyn-analyzers/src/Roslyn.Diagnostics.Analyzers/Setup/Roslyn.Diagnostics.Analyzers.Setup.csproj]

So I did what it asked :shrug: